pkg/networkmanager/firewall.jsx

```diff
@@ -1039,7 +1039,7 @@ export class Firewall extends React.Component {
         const enabled = this.state.firewall.enabled;
 
         return (
-            <Page groupProps={{ sticky: 'top' }}
+            <Page
```

Comment:
This commit should again start with a "where?", i.e. "networkmanager: Make firewall header not sticky". However, this again needs a "why", and it introduces an inconsistency: A lot of other page headers are sticky as well:
```
pkg/apps/application.jsx:        <Page groupProps={{ sticky: 'top' }}
pkg/networkmanager/firewall.jsx:            <Page groupProps={{ sticky: 'top' }}
pkg/networkmanager/network-interface.jsx:        <Page groupProps={{ sticky: 'top' }}
pkg/storaged/details.jsx:            <Page groupProps={{ sticky: 'top' }}
pkg/systemd/hwinfo.jsx:            <Page groupProps={{ sticky: 'top' }}
pkg/systemd/logDetails.jsx:            <Page groupProps={{ sticky: 'top' }}
pkg/systemd/services/service.jsx:                <Page groupProps={{ sticky: 'top' }}
pkg/users/account-details.js:        <Page groupProps={{ sticky: 'top' }}
```
and then there are others which aren't:
```
pkg/apps/application-list.jsx:        <Page id="list-page">
pkg/storaged/overview.jsx:        <Page id="main-storage">
```

Can you please discuss this with @garrett first? This should be done consistently on all pages. Perhaps this should also only be dropped on mobile? Having the breadcrumbs visible is useful, but indeed it's quite expensive in terms of screen size usage.

Reply:
Hello @martinpitt

Thanks for the recommendations! :) 
When the design is decided I will edit the commits/messages and add the new changes.

Reply:
Sticky can be problematic when it gets too large. Sticky is useful when there's some critical information or action at the top and it doesn't need much space. Sticky is probably not really useful just for breadcrumbs.

Additionally, sticky might or might not work based on other elements, and some pages may want to use sticky for other parts.

Changing sticky should be done in another PR, not this one. It's not related to fixing the padding. It was mentioned in the same issue, but as a question later. But it's a different type of change, so it should really be in a different PR.

Sometimes, we do have some somewhat unrelated changes in a PR as a separate commit in the same PR, like here, but it's usually for an umbrella of changes, and those PRs often take longer to merge. (In these cases, it's usually X, Y, Z commits for a PR on page A. So while X, Y, and Z changes might not be related to one another, they're still all related to A.)

But, in this PR as-is, it changes code that affects all of Cockpit (even projects) instead of fixing Applications and Kernel dump specifically. And then it has a sticky removal for Firewall which is not even closely related to Applications or Kernel dump or padding at all, so these two concepts really should be separate PRs.

## Summary

- Fix the problem on the problem pages, do not change something that affects everything (breaking everything else).
- Drop the sticky change here. Move it to another PR. We can talk about what pages should or should not have sticky and what parts over there on that other PR and adjust it accordingly.

## Hint

See this line that you commented out? `.pf-c-page__main > section.pf-c-page__main-section:not(.pf-m-padding)` — that `:not()` means that it won't apply when there's `pf-m-padding`. Sections don't have padding by default (as the widgets within sometimes may provide that). These two pages probably had padding as an accident before, but they don't now because they don't have widgets within that also provide padding _and_ these pages also didn't set padding.

So that means the fix would be to actually use widgets that have padding inside these headings _or_ to set padding. Look at what other pages (that don't have this problem) are doing.